src/core/server/root/index.ts

```diff
@@ -29,30 +29,26 @@ import { Server } from '../server';
  */
 export class Root {
   public readonly logger: LoggerFactory;
-  private readonly configService: ConfigService;
   private readonly log: Logger;
-  private readonly server: Server;
   private readonly loggingService: LoggingService;
+  private readonly server: Server;
   private loggingConfigSubscription?: Subscription;
 
   constructor(
     config$: Observable<Config>,
-    private readonly env: Env,
+    env: Env,
     private readonly onShutdown?: (reason?: Error | string) => void
   ) {
     this.loggingService = new LoggingService();
     this.logger = this.loggingService.asLoggerFactory();
     this.log = this.logger.get('root');
-
-    this.configService = new ConfigService(config$, env, this.logger);
-    this.server = new Server(this.configService, this.logger, this.env);
+    this.server = new Server(config$, env, this.logger);
   }
 
   public async setup() {
-    this.log.debug('setting up root');
-
     try {
-      await this.setupLogging();
+      this.log.debug('setting up root');
+      await this.setupLogging(this.server.configService);
```

Comment:
__nit:__ since these methods both share `this` context it doesn't seem necessary to pass argument since `setupLogging` can access `this.server` internally.